app/_utils/fetchLeaderboard.ts

```diff
@@ -0,0 +1,16 @@
+const baseUrl = process.env.NEXT_PUBLIC_BASE_URL;
```

Comment:
I would move this into the `getLeaderboardData`. I feel the top of the components should be limited to imports.